Bypass the DOM rendering queue if map is idle

zarov opened the pull request:
In #10530 I added a new queue dedicated to the rendering of DOM elements. Unfortunately I made the mistake to run this new queue only in `map._render()`, which won't be called if the map is idle. You can test this problem by trying to move a marker once the map is fully loaded and rendered in `markers.html`.

A mechanism like `map._requestRenderFrame()` could be added, but I feel there is no need to call the whole `_render()` loop all over again for a simple rendering of a DOM element. As far as I understand, this problem happens only when very few elements are moving/rendering. So I'm proposing to call directly the dom task, like it was the case before, only if the map is idle. The condition I choose is the same in `ui/map`, right before setting the map idle.

Note that preventing the map to go idle if there are still some tasks in `domRenderTaskQueue` won't work, as the queue could be filled while the map is already idle.

Tagging @mourner as he followed the previous PR.

Sorry about this mistake, I though I checked correctly in the other PR.

mourner reviewed (changes requested):
Since this code has to be repeated in multiple places, and it's easy to make this mistake again, let's add a `Map` method that contains this logic so that you do e.g. `this._map._addDomTask(() => { ... }` in either `marker.js`, `popup.js` and elsewhere.

zarov commented:
Good call, I added a `_requestDomTask` to `Map`. I also updated the affected components from the last time, except for `scale_control` where I am not sure how to proceed with the parameter to pass to `setScale` ([see here](https://github.com/mapbox/mapbox-gl-js/commit/925f9a3df411b1b6a96873672189af39f10f96f2#diff-274127645ee42974a7ee2fb9fcf978eb73193903d5671866878ce3fa3a5a769d)): replace `domQueue` by `map._requestDomTask` and call it in `setScale`, return the transformation callback from `setScale` and call it in `updateScale` or even something else?

mourner commented:
> except for `scale_control` where I am not sure how to proceed with the parameter to pass to `setScale`

In this case I think it's fine to go back to passing the `map` instance and using its method in `setScale`.

zarov commented:
Ok, I updated it!

karimnaaji commented:
Merging to have this in the rebased version of #10564.
